### PR TITLE
Add notebook gallery to the docs

### PR DESCRIPTION
**Pull request by mcflugen**
### Description

This pull request adds a notebook gallery to the docs. You can view the generated docs for this pull request on [readthedocs](https://landlab--1994.org.readthedocs.build/en/1994/).

The html generated from the notebooks does not contain output generated by any of the cells. This is because we store our notebooks with their output cleared. This is also the reason the notebook thumbnails have a default image of the *Landlab* ``L``. If we would like the docs to contain cell output, I think we could do something like the following,
* After we run the notebook tests in our GitHub workflow, upload the executed notebooks (which now contain output) as a build artifact.
* When running the docs workflow, download the artifacts first, then building the docs.
* Create an *htmlzip* file of the docs and upload that to *readthedocs* (we'll no longer build the docs on *readthedocs*).

If that's something we would like to explore, I would like to do so in a separate pull request.

### Checklist - did you ...

- [x] Add a [news fragment](https://landlab.readthedocs.io/en/master/development/contribution/index.html#news-entries) file entry if necessary?
- [x] Add / update tests if necessary?
- [x] Add new / update outdated documentation?
- [x] All tests have passed?
- [x] Formatted code with black?
- [x] Removed lint reported by flake8?
- [x] Sucessful documentation built? (if documentation added or modified)

### CODE REVIEW

**Review by mdpiper — APPROVED** [review]
I have opinions!

1. I'd like to see one big page with all the notebook thumbnails. Call it *Gallery* and have an entry for it in the left navbar.
2. On the main page, the link under the _Tutorials_ heading should still point to the eponymous section, but the text "Gallery of tutorial notebooks" should point to the *Gallery* page.

Maybe this is another PR?

**Comment by mcflugen** [comment]
@mdpiper Thanks for the opinions! I'll add a gallery page that shows all of the notebook thumbnails. Would you still like them grouped by topic (e.g. Advection, Agent Based Modeling, Boundary Conditions, etc.)?

What do you think about the heading cell for each of the notebooks? When the notebooks are viewed as a standalone notebook, it makes sense to have the first cell contain a Landlab logo, but when viewed within the context of our documention, it seems a little redundant (particularly because there's the exact same logo just to the left of the notebook. I'm think of removing the header cell from each of the notebooks. Or at least removing the Landlab logo from the cell. What do you think?

**Comment by mdpiper** [comment]
@mcflugen Yes and yes!

**Comment by mcflugen** [comment]
@mdpiper Could you please have another look and let me know what you think?

* All of the notebooks are now on one gallery page
* I removed the header image from all of the notebooks

File: docs/source/index.rst [hunk 1]
```diff
@@ -8,14 +8,34 @@ Landlab: A modular Earth Surface Dynamics modeling library
...
+Citing Landlab
+--------------
+
+If you use any portion of Landlab, please cite our :ref:`Citing Landlab <citing>`
+page.
+
+Tutorials
+---------
+
+:ref:`Gallery of tutorial notebooks <tutorials>`.
```
[the remark this answers is not in the document]
Comment (mcflugen) [reply docs/source/index.rst]: I've added those links.

**Comment by mcflugen** [comment]
I've also modified the GitHub actions *test.yml* workflow to include a job that builds the docs. Unlike how we build the docs in *docs.yml*, this new job includes the executed notebooks. This means the generated documentation will include output of notebook cells and the notebook thumbnails will contain the last images from notebooks.

This generates docs that look really cool but, unfortunately, I'm not sure what to do with it. At one time you could upload an *htmlzip* to *readthedocs* rather than building the documentation on their website but it seems that's not longer an option. We could upload it to *landlab.io*, I suppose.

**Comment by mdpiper** [comment]
> I've also modified the GitHub actions _test.yml_ workflow to include a job that builds the docs. Unlike how we build the docs in _docs.yml_, this new job includes the executed notebooks. This means the generated documentation will include output of notebook cells and the notebook thumbnails will contain the last images from notebooks.

Awesome!

> This generates docs that look really cool but, unfortunately, I'm not sure what to do with it. At one time you could upload an _htmlzip_ to _readthedocs_ rather than building the documentation on their website but it seems that's not longer an option. We could upload it to _landlab.io_, I suppose.

Could we host just the Gallery on landlab.github.io?

**Comment by mcflugen** [comment]
> Could we host just the Gallery on landlab.github.io?

We could. I like having the docs and the gallery all in one place, though. I think maybe a good option would be to move the complete documentation to [landlab.github.io](https://landlab.github.io) (this site needs to be updated, anyway, as it's out of date) but still have landlab.readthedocs.io functional for testing purposes (we could add a big banner to the top of each page). The two sites would be pretty much the same except the executed notebooks would be available on the new site.

Whatever we decide, let's do it in another pull request.

@mdpiper Do you think this pr is ready to go?

**Comment by mdpiper** [comment]
@mcflugen Sounds good to me! Merge away!

**Pull request merged**